app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CommentsController < ApplicationController
+
+  def new
+    @answer = Answer.find(params[:answer_id])
+    question = Question.find(params[:question_id])
+    if current_user
+      @comment = Comment.new(answer: @answer)
+    else
+      flash[:notice] = "You Mustache Login To Do That"
+      redirect_to question_path(question)
+    end
+  end
+
+  def create
+    answer = Answer.find(params[:answer_id])
+    if make_sure_that_user_logged_in
+      #add comment
+      user = current_user
+      params[:comment][:user_id] = user.id
+      comment = Comment.create(comment_params())
```

Comment:
user.comments.build / user.comments.create would obivate the need for modifying params here